### PR TITLE
[issue-126] Update default transaction timeouts in FlinkPravegaWriter to sync with Pravega defaults

### PR DESCRIPTION
**Pull request by vijikarthi**
Signed-off-by: Vijay Srinivasaraghavan <vijayaraghavan.srinivasaraghavan@emc.com>

**Change log description**
This PR addresses the issue https://github.com/pravega/flink-connectors/issues/126
- The transaction timeouts are updated to use the maximum defaults of Pravega

**Purpose of the change**
- To avoid flink app from failing due to larger transaction timeouts

**What the code does**
- configures default timeouts when the value is not supplied

**How to verify it**
`./gradlew clean build` should pass

### CODE REVIEW

File: src/main/java/io/pravega/connectors/flink/AbstractStreamingWriterBuilder.java [hunk 1]
```diff
@@ -24,8 +24,8 @@
 public abstract class AbstractStreamingWriterBuilder<T, B extends AbstractStreamingWriterBuilder> extends AbstractWriterBuilder<B> {
 
     // the numbers below are picked somewhat arbitrarily at this point
```
Comment (EronWright) [inline src/main/java/io/pravega/connectors/flink/AbstractStreamingWriterBuilder.java]: Is the above comment in the code still true?   Are these values consistent with the underlying Pravega client's defaults?  The new values are radically different, which suggests a problem, i.e. that we don't really understand them.

Comment (vijikarthi) [reply src/main/java/io/pravega/connectors/flink/AbstractStreamingWriterBuilder.java]: > // the numbers below are picked somewhat arbitrarily at this point

I forgot to remove this comment (will update). 

Yes, the values are consistent with Pravega's default and  I have posted my observation here https://github.com/pravega/flink-connectors/issues/126

File: src/main/java/io/pravega/connectors/flink/AbstractStreamingWriterBuilder.java [hunk 2]
```diff
@@ -23,9 +23,9 @@
  */
 public abstract class AbstractStreamingWriterBuilder<T, B extends AbstractStreamingWriterBuilder> extends AbstractWriterBuilder<B> {
 
-    // the numbers below are picked somewhat arbitrarily at this point
-    private static final long DEFAULT_TXN_TIMEOUT_MILLIS = 2 * 60 * 60 * 1000; // 2 hours
-    private static final long DEFAULT_TX_SCALE_GRACE_MILLIS = 10 * 60 * 1000; // 10 minutes
+    // the numbers below are picked based on the default max settings in Pravega
+    private static final long DEFAULT_TXN_TIMEOUT_MILLIS = 30000; // 30 seconds
```
Comment (fpj) [inline src/main/java/io/pravega/connectors/flink/AbstractStreamingWriterBuilder.java]: This timeout value corresponds to the lease period of a transaction. The writer renews the lease within the configured period, and it does it automatically, without any necessary intervention from the application. 30s sounds like a pretty reasonable default value to me.

This configuration parameter also determines the maximum amount of time we allow a txn to remain open. The maximum execution time is calculated as the minimum of 1 day and 1000x the lease period.

File: src/main/java/io/pravega/connectors/flink/AbstractStreamingWriterBuilder.java [hunk 3]
```diff
@@ -23,9 +23,9 @@
...
+    private static final long DEFAULT_TX_SCALE_GRACE_MILLIS = 30000; // 30 seconds
```
Comment (fpj) [inline src/main/java/io/pravega/connectors/flink/AbstractStreamingWriterBuilder.java]: The scale grace period is a parameter that is supposed to go away with the rolling txns feature. Currently, it cannot be smaller than the lease period.

**Comment by EronWright** [comment]
Here's the two options that I favor:

1. Rename `withTxnTimeout` to `withTxnLeaseRenewalPeriod`, default value of 30 seconds:
```
    /**
     * Sets the transaction lease renewal period.
     *
     * When the writer mode is set to {@code EXACTLY_ONCE}, transactions are used to persist events to the Pravega stream.  The transaction interval corresponds to the Flink checkpoint interval.   Throughout that interval, the transaction is kept alive with a lease that is periodically renewed.    This configuration setting sets the lease renewal period.  The default value is 30 seconds.
     *
     * @param period the lease renewal period
     */
    public B withTxnLeaseRenewalPeriod(Time period)
```

2. Keep the API as-is, change the default value to 8 hours, and divide the value by 1000 when configuring the underlying Pravega client (since it works in terms of lease renewal period).

**Comment by vijikarthi** [comment]
My suggestion is to merge this PR and create a separate issue post EAP since any change to the API will impact other dependencies.

**Comment by vijikarthi** [comment]
Can this be merged?

**Pull request merged**